src/pubsub/index.js

```diff
@@ -396,7 +396,7 @@ class PubsubBaseProtocol extends EventEmitter {
     if (msgs.length) {
       // @ts-ignore RPC message is modified
       msgs.forEach((message) => {
-        if (!(this.canRelayMessage || message.topicIDs.some((/** @type {string} */ topic) => this.subscriptions.has(topic)))) {
+        if (!(this.canRelayMessage || (message.topicIDs && message.topicIDs.some((topic) => this.subscriptions.has(topic))))) {
```

Comment:
oo, was this a bug that types picked up?

Reply:
This is because the protobuf definition does not state topicIDs as [required](https://github.com/libp2p/js-libp2p-interfaces/blob/v0.9.0/src/pubsub/message/rpc.proto.js#L16). As far as I know, all implementations send an empty arrau and not undefined and this has not been a problem. But could be as it is not spec'ed as required